src/pages/Search.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { getSearch } from '../data/API';
+import { useSearchParams } from 'react-router-dom';
+import SearchItem from '../components/search/SearchItem';
+import style from './Search.module.css';
+
+const Search = () => {
+  const [search, setSearch] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  // location 의 ?s=title 가져오기
+  let [searchParams, setSearchParams] = useSearchParams();
+  let title = searchParams.get('s');
+
+  const TAGS = [
+    '농산물',
+    '과일',
+    '채소',
+    '곡물',
+    '수산물',
+    '제철',
+    '건어물',
+    '축산물',
+    '가공식품',
+    '일반',
+    '반찬',
+    '차류',
+    '벌꿀',
+    '간편간식',
+  ];
+
+  // 키워드가 두개이상 일 때 구분
+  let titleArr;
+  let tag = TAGS.find((tag) => title.includes(tag));
+  let findTitle;
+  if (title.includes(' ')) {
+    titleArr = title.split(' ');
+    tag = TAGS.find((tag) => titleArr.includes(tag));
+    findTitle = titleArr.find((t) => t !== tag);
+  } else if (!title.includes(' ') && tag) {
+    title = tag;
+  }
+
+  // 구분해서 search data 넣기
+  useEffect(() => {
+    if (findTitle && tag) {
+      const searchData = getSearch(findTitle, tag);
+      searchData.then((data) => {
+        setSearch(data);
+        setLoading(false);
+      });
+    } else if (!findTitle && tag) {
+      const searchData = getSearch('', tag);
+      searchData.then((data) => {
+        setSearch(data);
+        setLoading(false);
+      });
+    } else {
+      const searchData = getSearch(title);
+      searchData.then((data) => {
+        setSearch(data);
+        setLoading(false);
+      });
+    }
+  }, []);
+
+  const filters = ['정확도순', '낮은 가격순', '높은 가격순'];
+  const [filter, setFilter] = useState(filters[0]);
+  const filtered = getFilteredProducts(filter, search)
+
+  function getFilteredProducts(filter, search) {
+    if(filter === '정확도순') {
+      return search;
+    } else if(filter === '낮은 가격순') {
+      let copyLow = [...search]
+      copyLow.sort(function(a,b){
+          return a.price - b.price;
+        })
+      return copyLow;
+    } else if(filter === '높은 가격순') {
+      let copyHigh = [...search]
+      copyHigh.sort(function(a,b){
+        return b.price - a.price;
+      })
+      return copyHigh;
+    }
+  }
```

Comment:
```suggestion
const getFilteredProducts = (filter, search) => {
  let copy = [...search];
  if (filter === "낮은 가격순") {
    copy.sort((a, b) => a.price - b.price);
  }
  if (filter === "높은 가격순") {
    copy.sort((a, b) => b.price - a.price);
  }
  return copy;
};
```
함수를 이렇게 정의해도 같은 역할을 합니다.
1. if 문에 항상 else 를 붙일 필요는 없습니다.
2. 반복되는 코드를 줄일 수 있습니다.
3. 각 조건분기마다 무슨 값을 리턴하는 것보다.. 분기별로 데이터를 가공하고, 마지막에 리턴코드를 하나만 넣는게 더 가독성이 좋을 수 있습니다.